frames/src/main/java/com/checkout/frames/component/cardholdername/CardHolderNameViewModel.kt

```diff
@@ -40,23 +40,25 @@ internal class CardHolderNameViewModel @Inject constructor(
 
         if (!isFocused && wasFocused) {
             val cardHolderName = componentState.cardHolderName.value
-            componentState.cardHolderName.value = cardHolderName
-            handleValidationResult(cardHolderName)
+            handleValidationResult(cardHolderName, true)
         }
     }
 
-    private fun handleValidationResult(cardHolderName: String) {
+    private fun handleValidationResult(cardHolderName: String, isRequiredToHandleError: Boolean) {
         if (cardHolderName.isEmpty() && !componentStyle.isInputFieldOptional) {
```

Comment:
nit: we can simply this block to this? so that it's easier to understand the `update { value }` part?

```kotlin
val isValid = cardHolderName.isNotEmpty() || componentStyle.isInputFieldOptional
paymentStateManager.isCardHolderNameValid.update { isValid }

if (isRequiredToHandleError) when {
    isValid -> showValidationError()
    else -> hideValidationError()
}
```